internal/service/sesv2/dedicated_ip_assignment.go

```diff
@@ -0,0 +1,172 @@
+package sesv2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sesv2"
+	"github.com/aws/aws-sdk-go-v2/service/sesv2/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func ResourceDedicatedIPAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceDedicatedIPAssignmentCreate,
+		ReadWithoutTimeout:   resourceDedicatedIPAssignmentRead,
+		DeleteWithoutTimeout: resourceDedicatedIPAssignmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"destination_pool_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
I opted for `ForceNew` here, even though technically the IP could be moved via an update. This felt cleaner for the case where a config like the following has a `pool_name` change:

```terraform
resource "aws_sesv2_dedicated_ip_pool" "test" {
  pool_name = "my-pool" # ex. change to my-pool2
}

resource "aws_sesv2_dedicated_ip_assignment" "test" {
  ip                    = "0.0.0.0"
  destination_pool_name = aws_sesv2_dedicated_ip_pool.test.pool_name
}
```

Without `ForceNew`:
- `apply` attempts to delete `aws_sesv2_dedicated_ip_pool.test`, but fails because the pool isn't empty. This is resolved by a two-step apply. First, removing the assignment resource and applying to destroy it. Then the pool name can be updated and the assignment safely added back.
  - Adding logic to the dedicated IP pool destroy function to empty it first can't be done safely because it will cause a state change for assignment resources mid-apply.

With `ForceNew`:
- Both resources are destroyed and re-created with no side effects.